### PR TITLE
Feature: Automate new component creation

### PR DESCRIPTION
**Pull request by thomasdigby**
- Create a new utility to provide a step by step creation of a new component or utility
- Choose type of component (Component or Utility), the component name and the category (for documentation)
- It will create an `index.ts`, `Component.tsx`, `Component.test.tsx` and `Component.mdx`

https://user-images.githubusercontent.com/3226804/111365080-f4c31800-8689-11eb-8e31-86462b4ba02b.mov

### CODE REVIEW

File: scripts/add-component.mjs [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  if (confirmation && !fs.existsSync(directory)) {
+    // safe to do as we know the directory does not exist
+    fs.mkdirSync(directory)
+    // go and write files from templates with content
+    await Promise.all([
+      writeFile('index.ts', templates.index, response),
+      writeFile(`${name}.tsx`, templates.component, response),
+      writeFile(`${name}.test.tsx`, templates.test, response),
+      writeFile(`${name}.mdx`, templates.documentation, response)
+    ])
```
Comment (andoulla) [inline scripts/add-component.mjs:174]: Very tiny, but maybe a `catch` here would be a nice addition as we can handle how the error is displayed

File: scripts/add-component.mjs [hunk 2]
```diff
@@ -0,0 +1,182 @@
+// eslint-disable-next-line
+import { default as changeCase } from 'change-case'
```
Comment (hollg) [inline scripts/add-component.mjs:2]: Is this different to `import changeCase from 'change-case'`?

**Comment by hollg** [comment]
This is _so cool_!

**Review by hollg — APPROVED** [review]

File: scripts/add-component.mjs (hunk 2 above)
Comment (thomasdigby) [reply scripts/add-component.mjs:2]: I was having a few issues with this import as `change-case` doesn't technically have a default import, node is shimming it. Reckon we can go back to that import statement though 👍

**Pull request merged**